src/util/simplify_expr2.cpp

```diff
@@ -415,13 +414,87 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // (A + 1) + ~B --> A - B
+  if(is_add2t(side_1) && is_bitnot2t(side_2))
+  {
+    dump();
+    auto sidecheck = to_add2t(side_1).side_2;
+    auto sidecheck2 = to_add2t(side_1).side_1;
+    if(is_constant_int2t(sidecheck) == 1)
+    {
+      printf("ENTRADA------------------------------------------\n");
+      dump();
+      auto B = to_bitnot2t(side_2).value;
+      auto new_operand = sub2tc(side_1->type, sidecheck2, B);
+      printf("SAIDA------------------------------------------\n");
+      new_operand->dump();
+      return new_operand;
+    }
+  }
+  // (~B + A) + 1 --> A - B
+  //&&
+  // (A + ~B) + 1 --> A - B
+  if(is_add2t(side_1) && is_constant_int2t(side_2) == 1)
```

Comment:
is_constant_int returns a boolean value.

If you want to check if the value itself is 1, then use `to_constant_int(side_2).value`

Reply:
tks i last this one pass through I changed this one for   if(is_add2t(side_1) && is_constant_int2t(side_2) && to_constant_int2t(side_2).value == 1) too